acceptance/acceptance_test.go

```diff
@@ -806,15 +820,24 @@ func runWithLog(t *testing.T, cmd *exec.Cmd, out *os.File, tail bool) error {
 	r, w := io.Pipe()
 	cmd.Stdout = w
 	cmd.Stderr = w
+	processErrCh := make(chan error, 1)
+
+	cmd.Cancel = func() error {
+		processErrCh <- errors.New("Test script killed due to a timeout")
+		_ = cmd.Process.Kill()
+		_ = w.Close()
+		return nil
+	}
 
 	start := time.Now()
 	err := cmd.Start()
-	require.NoError(t, err)
+	if err != nil {
+		return err
+	}
 
-	processErrCh := make(chan error, 1)
 	go func() {
 		processErrCh <- cmd.Wait()
```

Comment:
I suspect `cmd.Wait()` returns some time after `cmd.Process.Kill()` returns.

This means we'll see two errors on `processErrCh`.

Given the return statement consumes one, there is room for the other and this goroutine will terminate. But from a readability pov, it makes sense to buffer 2 messages on the channel. There are 2 writers, after all.

Reply:
> I suspect cmd.Wait() returns some time after cmd.Process.Kill() returns.

yes, it should.

> This means we'll see two errors on processErrCh.

yes, although we only read the first one, like you noted. The second one is ignored. I'd be concerned if it means second goroutine hangs for some prolonged time, but in either case it should terminate right away.

> But from a readability pov, it makes sense to buffer 2 messages on the channel.

Could you elaborate, is it readability of the code or output of test runner?  (I assume first). What do you suggest to change here?

Reply:
The second goroutine will hang until one message is consumed from the channel.

By allocating space for 2 messages on the channel, it is clear that neither goroutine will ever hang. You're right that this is not critical, of course, but a comment explaining the nuance or expected flow can help.